### ISSUE TITLE
Add missing optimizer attributes to Python interface

### ISSUE THREAD

**Issue by bmhowe23** [NVIDIA/cuda-quantum#2403]
Hi @anandaroy7 - are you using C++ or Python? If you're using C++, I think you can use the `f_tol` and `step_size` attributes. I am not sure if these are exposed to Python at the moment, but if they are missing, we could certainly add them.

_Originally posted by @bmhowe23 in https://github.com/NVIDIA/cuda-quantum/discussions/2329#discussioncomment-11102211_

**Comment by Squirtle007** [comment]
Same Here. 

Additional notes:
It would be helpful if intermediate results could be printed out. Inspired by the `callback` in `minimize` from [SciPy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html), an optional setting helps keep track of convergence for developers.

**Comment by Squirtle007** [comment]
Bringing this up again since many developers face the same problem.

Specifically, `step_size` (learning rate) is a critical hyperparameter for gradient-based optimizers such as `Adam` and `SGD`.
But for now, this doesn't work:
```
optimizer = cudaq.optimizers.Adam()
optimizer.step_size = 0.1
```
Could we expose the `step_size` attribute to Adam/SGD in Python?
Thanks!

**Comment by hungsheng8899** [comment]
me too

**Comment by Squirtle007** [comment]
Hi @bmhowe23 , could someone please look into this small but relevant feature? Thanks in advance.

**Comment by bmhowe23** [comment]
@sacpis - any chance you could jump in here?

**Comment by sacpis** [comment]
Sure, happy to look at it.

**Comment by sacpis** [comment]
@bmhowe23 Is this issue good to close now?

**Comment by sacpis** [comment]
Sorry, accidentally hit the Close issue button. Reopened it.

**Comment by sacpis** [comment]
@bmhowe23 Please feel free to close this issue when you have a chance.

**Comment by Squirtle007** [comment]
Thanks @sacpis , and sorry for the late feedback - it has been tested and works well. Had included it in the workshop tutorials.
Thanks again.
(recommend closing this, add @bmhowe23 for viz)
